crates/ty_python_semantic/src/lint.rs

```diff
@@ -344,9 +344,9 @@ impl LintRegistryBuilder {
             ),
         };
 
+        let entry = LintEntry::Alias(*target);
         assert_eq!(
-            self.by_name
-                .insert(from.as_str(), LintEntry::Alias(*target)),
+            self.by_name.insert(from.as_str(), entry),
```

Comment:
What's the reason for this change?

Reply:
Using `Deref` prevented [two-phase borrows](https://rustc-dev-guide.rust-lang.org/borrow_check/two_phase_borrows.html) from working properly. Now that we're no longer using `Deref`.